api/server/server.go

```diff
@@ -1103,50 +1100,33 @@ func (s *Server) postContainersAttach(version version.Version, w http.ResponseWr
 		return fmt.Errorf("Missing parameter")
 	}
 
-	cont, err := s.daemon.Get(vars["name"])
-	if err != nil {
-		return err
-	}
-
 	inStream, outStream, err := hijackServer(w)
 	if err != nil {
 		return err
 	}
 	defer closeStreams(inStream, outStream)
 
-	var errStream io.Writer
-
 	if _, ok := r.Header["Upgrade"]; ok {
 		fmt.Fprintf(outStream, "HTTP/1.1 101 UPGRADED\r\nContent-Type: application/vnd.docker.raw-stream\r\nConnection: Upgrade\r\nUpgrade: tcp\r\n\r\n")
 	} else {
 		fmt.Fprintf(outStream, "HTTP/1.1 200 OK\r\nContent-Type: application/vnd.docker.raw-stream\r\n\r\n")
 	}
 
-	if !cont.Config.Tty && version.GreaterThanOrEqualTo("1.6") {
-		errStream = stdcopy.NewStdWriter(outStream, stdcopy.Stderr)
-		outStream = stdcopy.NewStdWriter(outStream, stdcopy.Stdout)
-	} else {
-		errStream = outStream
+	attachWithLogsConfig := &daemon.ContainerAttachWithLogsConfig{
+		InStream:   inStream,
+		OutStream:  outStream,
+		UseStdin:   boolValue(r, "stdin"),
+		UseStdout:  boolValue(r, "stdout"),
+		UseStderr:  boolValue(r, "stderr"),
+		Logs:       boolValue(r, "logs"),
+		Stream:     boolValue(r, "stream"),
+		ApiVersion: version,
```

Comment:
@vieux this was one of my concern as well but since the stream configuration depends also on apiversion I couldn't do other way (we did the same for api logs call)